### ISSUE TITLE
Agenda Fealty didn't reduce cost in Challenges phase

### ISSUE THREAD

**Issue by therealyan** [throneteki/throneteki#2338]
Playing with the House Greyjoy I won an unopposed challenge, so I wanted to play the We Do Not Sow event card. Since I had no gold left, I decided to use the Fealty agenda card to reduce the cost of the next loyal card I'm playing by 1 (We Do Not Sow costs exactly 1 gold). The chatbox printed out the message that I'm reducing the cost of the next card played, however I couldn't select the event card (it was grayed out).

Is it a bug, or am I missing something?

**Comment by ystros** [comment]
Thanks for the report! I've been able to reproduce this bug once or twice, but it seems very inconsistent - sometimes Fealty w/ 0 gold allows a 1 cost event, sometimes not. Have you seen this again since your report? Have you noticed any patterns?

**Comment by therealyan** [comment]
Yeah, so far it worked only once for me: when opponent declares defenders (in that case it was 0 defenders), I need to kneel faction card via Fealty. After that during the next action window (before applying claim) I can trigger We Do Not Sow event for 0 gold.
If I miss the action window after defenders were declared (do not kneel faction card at this moment), and try to kneel it before applying claim (when the result of the challenge is being printed in the log), the log says that Fealty is being used outside of action window, and it doesn't allow me to play We Do Not Sow event for 0 gold.
I'll try to do more testing and will post results if I find more patterns.
